src/main/scala/io/iohk/ethereum/consensus/ethash/validators/EthashBlockHeaderValidator.scala

```diff
@@ -41,11 +42,11 @@ class EthashBlockHeaderValidator(blockchainConfig: BlockchainConfig)
 
     import scala.collection.JavaConverters._
 
-    def getPowCacheData(epoch: Long): PowCacheData = {
+    def getPowCacheData(epoch: Long, seed: ByteString): PowCacheData = {
       Option(powCaches.get(epoch)) match {
         case Some(pcd) => pcd
         case None =>
-          val data = new PowCacheData(cache = EthashUtils.makeCache(epoch), dagSize = EthashUtils.dagSize(epoch))
+          val data = new PowCacheData(cache = EthashUtils.makeCache(epoch, seed), dagSize = EthashUtils.dagSize(epoch))
 
           val keys = powCaches.keySet().asScala
           val keysToRemove = keys.toSeq.sorted.take(keys.size - MaxPowCaches + 1)
```

Comment:
I wonder if it won't make problems during transition. i.e

1. lets say we have epoch `358` and `359` here. 
2. we enter fork block `ecip-1099` so next epoch is `180` instead of `360`
3. so we put dag cache for this epoch in cache, and we have there dag for epoch `180` and `359`  (as we have limit of 2 dag caches max, and we always removes oldest one)
4. after some blocks we arive at epoch `181`. And now we will remove dag cache for epoch `180` instead on `359` .  
It means dag dag cache for epoch `359` will be with us for long time (until we get to new epoch `360`), and if rollbacks happen accross new epoch boundaries we will need to recreate dag caches. (we have one stale in there, and we have max size of 2 dag caches )